generator/src/main.rs

```diff
@@ -189,9 +167,7 @@ async fn main() -> Result<()> {
             true,
             &out_root,
         )?;
-    }
-
-    if let Some(m) = &on_chain_model {
+    } else if let Some(m) = &on_chain_model {
```

Comment:
You introduced a bug here because now the on chain dependency graph and top level packages won't be generated if we have the source model. We actually want both so this should be an `if` instead of `else if`.

There's two separate dependnecy graphs because I wanted to support on-chain packges but when I started this there wasn't a way to consolidate source and on-chain deps under the same graph. I was then waiting for on-chain deps work to land on master but since this is not happening yet I think I will just use the on-chain hooks I implemented for that here.